app/uk/gov/hmrc/apprenticeshiplevy/controllers/sandbox/data/SandboxTestDataController.scala

```diff
@@ -19,40 +19,78 @@ package uk.gov.hmrc.apprenticeshiplevy.controllers.sandbox.data
 import java.io.{File, FileInputStream, InputStream}
 import java.net.URLDecoder
 
+import com.google.inject.{Inject, Singleton}
 import org.joda.time._
 import org.slf4j.MDC
 import play.api.libs.json._
 import play.api.mvc._
-import play.api.{Logger, Mode}
+import play.api.{Configuration, Logger, Mode}
 import uk.gov.hmrc.apprenticeshiplevy.config.AppContext
 import uk.gov.hmrc.play.microservice.controller.Utf8MimeTypes
 
 import scala.concurrent.Future
 import scala.io.Source
 import scala.util.Try
 
-class SandboxTestDataController extends Controller with Utf8MimeTypes {
+@Singleton
+class SandboxTestDataController @Inject()(configuration: Configuration) extends Controller with Utf8MimeTypes {
   val SANDBOX_DATA_DIR = "public/sandbox-data"
 
+  lazy val returnDummyResponse: Boolean =
+    configuration
+      .getBoolean("features.returnDummyResponse")
+      .getOrElse(false)
+
+  private[sandbox] def getFileName(file: String) =
+    if(returnDummyResponse) {
+      val empRefRegex = "^.*employers?\\/([a-zA-Z0-9]{3}\\/.*?|.*?)\\/.*".r
+      file match {
+        case empRefRegex(empRef) =>
+          Logger.debug(s"Replacing empref: $empRef in requested file: $file")
+          val dummyEmpRef = List("840","MODES17")
+          val ninoRegex = "^.*\\/employed\\/(.{8,9})\\?.*$".r
+
+          val updatedFile = file.replace(empRef,
```

Comment:
Nested if could be read easier if it were a val. Preference I guess though.

Reply:
I have pulled the if into a val